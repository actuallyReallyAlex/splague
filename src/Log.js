import React from 'react'
import { Comment, Header } from 'semantic-ui-react'
import PropTypes from 'prop-types'
import LogItem from './LogItem'

const logStyle = {
  color: 'white'
}

const Log = props => {
  return (
    <Comment.Group>
      <Header as="h3" style={logStyle}>
        Log
      </Header>

      {Object.keys(props.log).map((logItem, i) => {
        return (
          <LogItem
            key={i}
            author="Plague"
            metaData="Day 0"
            text={props.log[logItem]}
          />
        )
      })}
    </Comment.Group>
  )
}

Log.propTypes = {
  log: PropTypes.object
}

export default Log
