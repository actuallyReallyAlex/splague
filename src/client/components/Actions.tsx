import * as React from "react";
import { connect } from "react-redux";
import { performAction } from "../redux/thunks";
import { LocationAction, RootState } from "../types";

export interface ActionsProps {
  actions: LocationAction[];
  handlePerformAction: (action: LocationAction) => void;
}

const Actions: React.SFC<ActionsProps> = (props: ActionsProps) => {
  const { actions, handlePerformAction } = props;
  return (
    <div id="actions">
      {actions.map((action: LocationAction) => (
        <button
          className="nes-btn"
          key={action}
          id={`action-${action.replace(/ /gm, "-")}`}
          onClick={() => handlePerformAction(action)}
        >
          {action}
        </button>
      ))}
    </div>
  );
};

const mapStateToProps = (state: RootState) => ({ actions: state.map.actions });

const mapDispatchToProps = (dispatch) => ({
  handlePerformAction: (action: LocationAction) =>
    dispatch(performAction(action)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Actions);
