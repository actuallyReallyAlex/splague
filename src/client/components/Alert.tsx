import * as React from "react";
import { connect } from "react-redux";
import { RootState, Theme } from "../types";

export interface AlertProps {
  content: string;
  isOpen: boolean;
  primaryAction: () => void;
  primaryActionText: string;
  secondaryAction: () => void;
  secondaryActionText: string;
  theme: Theme;
  title: string;
}

const Alert: React.SFC<AlertProps> = (props: AlertProps) => {
  const {
    content,
    isOpen,
    primaryAction,
    primaryActionText,
    secondaryAction,
    secondaryActionText,
    theme,
    title,
  } = props;
  return (
    <div className={`dialog-container ${isOpen ? "" : "hidden"}`}>
      <dialog
        className={`dialog-inner nes-dialog ${
          theme === "dark" ? "is-dark" : ""
        }`}
        id="alert"
      >
        <form method="dialog">
          <p className="title">{title}</p>
          <p>{content}</p>
          <menu className="dialog-menu">
            <button
              className="nes-btn is-primary"
              id="alert-primary"
              onClick={() => primaryAction()}
            >
              {primaryActionText}
            </button>
            {secondaryActionText && (
              <button
                className="nes-btn"
                id="alert-secondary"
                onClick={() => secondaryAction()}
              >
                {secondaryActionText}
              </button>
            )}
          </menu>
        </form>
      </dialog>
    </div>
  );
};

const mapStateToProps = (state: RootState) => ({
  content: state.alert.content,
  isOpen: state.alert.isOpen,
  primaryAction: state.alert.primaryAction,
  primaryActionText: state.alert.primaryActionText,
  secondaryAction: state.alert.secondaryAction,
  secondaryActionText: state.alert.secondaryActionText,
  theme: state.ui.theme,
  title: state.alert.title,
});

export default connect(mapStateToProps)(Alert);
