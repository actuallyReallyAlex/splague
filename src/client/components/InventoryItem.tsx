import * as React from "react";
import { InventoryItem } from "../types";

export interface InventoryItemProps {
  item?: InventoryItem;
}

const InventoryItem: React.SFC<InventoryItemProps> = (
  props: InventoryItemProps
) => {
  const { item } = props;
  return (
    <td data-tip={item ? item.name : ""} className="inventory-slot">
      {item && <img alt={item.name} src={item.image} />}
    </td>
  );
};

export default InventoryItem;
