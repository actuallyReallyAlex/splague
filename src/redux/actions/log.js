export const ADD_LOG_ITEM = 'ADD_LOG_ITEM'

/**
 * Adds a log item to the Log.
 * @param {Object} logItem LogItem details.
 */
export const addLogItem = logItem => {
  return {
    type: ADD_LOG_ITEM,
    payload: {
      logItem
    }
  }
}
