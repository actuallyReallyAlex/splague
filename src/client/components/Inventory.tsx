import * as React from "react";
import { connect } from "react-redux";
import { setInventoryIsOpen } from "../redux/actions/inventory";
import { RootState, Theme } from "../types";

export interface InventoryProps {
  handleCloseInventory: () => void;
  isOpen: boolean;
  theme: Theme;
}

const Inventory: React.SFC<InventoryProps> = (props: InventoryProps) => {
  const { handleCloseInventory, isOpen, theme } = props;
  return (
    <div className={`dialog-container ${isOpen ? "" : "hidden"}`}>
      <dialog
        className={`dialog-inner nes-dialog ${
          theme === "dark" ? "is-dark" : ""
        }`}
        id="inventory"
      >
        <form method="dialog">
          <p className="title">Inventory</p>
          <menu className="dialog-menu">
            <button
              className="nes-btn is-primary"
              id="close-inventory"
              onClick={() => handleCloseInventory()}
            >
              Close
            </button>
          </menu>
        </form>
      </dialog>
    </div>
  );
};

const mapStateToProps = (state: RootState) => ({
  isOpen: state.inventory.isOpen,
  theme: state.ui.theme,
});

const mapDispatchToProps = (dispatch) => ({
  handleCloseInventory: () => dispatch(setInventoryIsOpen(false)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Inventory);
