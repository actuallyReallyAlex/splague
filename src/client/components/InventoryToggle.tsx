import * as React from "react";
import { connect } from "react-redux";
import { setInventoryIsOpen } from "../redux/actions/inventory";

export interface InventoryToggleProps {
  handleInventoryToggle: () => void;
}

const InventoryToggle: React.SFC<InventoryToggleProps> = (
  props: InventoryToggleProps
) => {
  const { handleInventoryToggle } = props;
  return (
    <button
      className="nes-btn is-primary"
      id="open-inventory"
      onClick={() => handleInventoryToggle()}
    >
      Inventory
    </button>
  );
};

const mapDispatchToProps = (dispatch) => ({
  handleInventoryToggle: () => dispatch(setInventoryIsOpen(true)),
});

export default connect(null, mapDispatchToProps)(InventoryToggle);
