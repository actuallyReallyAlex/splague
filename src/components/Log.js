import React, { Component } from 'react'
import PropTypes from 'prop-types';
import { connect } from 'react-redux'
import { Box, Button } from 'grommet'
import LogItem from './LogItem'
import { addLogItem } from '../redux/actions/actions'

class Log extends Component {
  static propTypes = {
    log: PropTypes.array
  }
  static defaultProps = {
    log: []
  }
  
  componentDidMount() {
    const { dispatch } = this.props
    setTimeout(() => {
      dispatch(addLogItem({ title: 'lolol', description: 'buttt' }))
    }, 3000)
  }

  render() {
    const { dispatch, log } = this.props
    return (
      <Box
        background="#424344"
        fill="vertical"
        gap="small"
        pad="small"
        width="300px"
      >
        {log.map((item, index) => (
          <LogItem {...item} key={`log-${index}`} />
        ))}
        <Button
          label="Test"
          onClick={() =>
            dispatch(addLogItem({ title: 'test', description: 'testy' }))
          }
        />
      </Box>
    )
  }
}

const mapStateToProps = ({ log }) => {
  return { log }
}

export default connect(mapStateToProps)(Log)
