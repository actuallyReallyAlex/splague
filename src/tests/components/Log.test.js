import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { Log } from '../../components/Log'
import log from '../fixtures/log'

const dispatch = jest.fn()

describe('<Log />', () => {
  it('Should render snapshot of Log component.', () => {
    const component = renderer
      .create(<Log dispatch={dispatch} log={log} />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
