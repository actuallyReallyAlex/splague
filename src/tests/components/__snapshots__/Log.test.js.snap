// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Log /> Should render snapshot of Log component. 1`] = `
.c3 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #FFFFFF;
  stroke: #FFFFFF;
}

.c3 g {
  fill: inherit;
  stroke: inherit;
}

.c3 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c3 *[stroke*="#"],
.c3 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c3 *[fill-rule],
.c3 *[FILL-RULE],
.c3 *[fill*="#"],
.c3 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  width: 15%;
  background: #424344;
  color: #f8f8f8;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  padding: 12px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  background: #FFFFFF;
  color: #444444;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 12px;
  border-radius: 12px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  height: 50px;
  width: 50px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #000000;
  color: #f8f8f8;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  border-radius: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  outline: none;
  max-width: 100%;
  margin: 0px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c4 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 12px;
}

.c9 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 12px;
}

.c10 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  outline: none;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #6FFFB0;
  border-radius: 18px;
  color: #f8f8f8;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  -webkit-transition: 0.1s ease-in-out;
  transition: 0.1s ease-in-out;
}

.c10:hover {
  box-shadow: 0px 0px 0px 2px #6FFFB0;
}

.c6 {
  font-size: 18px;
  line-height: 24px;
  font-weight: bold;
}

.c7 {
  margin: 0px;
  font-size: 14px;
  line-height: 20px;
  margin-top: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 0px;
  color: grey;
}

.c8 {
  font-size: 14px;
  line-height: 20px;
}

@media only screen and (max-width:768px) {
  .c0 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    border-radius: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c2 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    margin: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    height: 6px;
  }
}

<div
  className="c0"
>
  <div
    className="animated fadeInUp c1"
  >
    <div
      className="c2"
    >
      <svg
        aria-label="Alert"
        className="c3"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,17 L12,19 M12,10 L12,16 M12,3 L2,22 L22,22 L12,3 Z"
          fill="none"
          stroke="#000"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="c4"
    />
    <div
      className="c5"
    >
      <span
        className="c6"
      >
        Test Item 1
      </span>
      <span
        className="c7"
        size="small"
      >
        Additional Info
      </span>
      <span
        className="c8"
        size="small"
      >
        Test Description
      </span>
    </div>
  </div>
  <div
    className="c9"
  />
  <button
    className="c10"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    type="button"
  >
    Test
  </button>
</div>
`;
